libevmasm/AssemblyItem.cpp

```diff
@@ -88,9 +88,11 @@ size_t AssemblyItem::bytesRequired(size_t _addressLength) const
 		return 1 + 32;
 	case AssignImmutable:
 		if (m_immutableOccurrences)
-			return 1 + (3 + 32) * *m_immutableOccurrences;
+			// (DUP DUP PUSH <n> ADD MSTORE)* (PUSH <n> ADD MSTORE)
+			return (*m_immutableOccurrences - 1) * (5 + 32) + (3 + 32);
 		else
-			return 1 + (3 + 32) * 1024; // 1024 occurrences are beyond the maximum code size anyways.
+			// POP POP
```

Comment:
The condition above does not check for zero, but for "known".

Reply:
Now taken into account (above).